packages/core/sdk/src/internal/net/BasicRequest.ts

```diff
@@ -85,9 +85,11 @@ export namespace BasicRequest {
       return new BasicRequest(
         this.method,
         this.url,
+        /* tslint:disable no-object-literal-type-assertion */
```

Comment:
[nit] add comment explaining why you disabled the rule/why we can't resolve this by making the types better